src/Plugin/Field/FieldWidget/OgComplex.php

```diff
@@ -76,7 +69,147 @@ public function elementValidate($element, FormStateInterface $form_state, $form)
       $form_state->setError($element, t('The entity %label is not a group.', $params));
       return;
     }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(FieldItemListInterface $items, array &$form, FormStateInterface $form_state, $get_delta = NULL) {
+    $parent_form = parent::form($items, $form, $form_state, $get_delta);
+
+    $parent_form['other_groups'] = [];
+
+    // Adding the other groups widget.
+    $this->otherGroupsWidget($parent_form['other_groups'], $form_state);
+    return $parent_form;
+  }
+
+  /**
+   * Adding the other groups widget to the form.
+   *
+   * @param $elements
+   *   The widget array.
+   */
+  private function otherGroupsWidget(&$elements, FormStateInterface $form_state) {
```

Comment:
Let's go with protected, in case people want to ~~fuck with it~~ extend this class.